dashboard/src/actions/authActions.js

```diff
@@ -1,56 +1,63 @@
 import * as TYPES from "./types";
 import API from "../utils/api";
-import * as API_ROUTES from "../utils/apiConstants";
 import Cookies from "js-cookie";
 import { uid } from "../utils/helper";
 
-export const makeLoginRequest = (details, navigate) => async (dispatch, getState) => {
-  try {
-    dispatch({ type: TYPES.LOADING });
-    //empty the alerts
-    dispatch({
-      type: TYPES.USER_NOTION_ALERTS,
-      payload: [],
-    });
-    const response = await API.post(API_ROUTES.LOGIN, {
-      ...details,
-    });
-    if (response.status === 200 && Object.keys(response.data).length > 0) {
-      let keepUser = getState().userAuth.keepLoggedIn;
-      let expiryTime = keepUser ? 7 : 0.5;
-      Cookies.set("isLoggedIn", true, { expires: expiryTime, secure:true });
-      Cookies.set("token", response.data?.auth_token, { expires: expiryTime, secure:true });
-      Cookies.set("username", response.data?.username, { expires: expiryTime, secure:true });
-      navigate("/");
+export const makeLoginRequest =
+  (details, navigate) => async (dispatch, getState) => {
```

Comment:
I think this is the standard way to dispatch an action. In case any changes needed, let's dig into it later.

Reply:
I think this is my biggest source of discomfort with the Javascript convention of binding names to lambdas instead of just declaring functions: it causes patterns like this which are awkward to follow visually.

And, yeah; binding a lambda to another lambda just seems odd.

Reply:
@dbutenhof, Varshini tossed me a clue in her other PR:  it's a _thunk_.  Apparently, it's a pretty common technique in Javascript used to defer the construction of values:  instead of constructing them in-line, you create a function which provides the value and then you reference the function when you want the value.  The usefulness of this is obvious if the value is produced asynchronously, but it's also useful to avoid producing complex values when they aren't actually used.  So, I understand and appreciate it as a concept.

However, like any tool, I would recommend only using this idiom when it's actually needed or otherwise beneficial.  I would recommend against using it just because it has become a habit.  It is intended to be a tradeoff between making obtaining the value more expensive (by wrapping it inside a function call) against avoiding producing the value initially (while rendering the page) or at all (if it ends up never being needed, because the user never clicked in the right spot).  The problem is, if producing the value is less expensive than the function call required to get it, then we should avoid using a thunk!

As for the distinction between using the `function` keyword and the `=>` thingie (apparently, it is _not_ an "operator"...does that make it a "keyword"??), there doesn't seem to be a huge difference, and so I guess I can get used to what Dave is calling "binding names to lambdas".  But, in a situation like the one here, where what we are doing is defining a named function, I would definitely lean toward using the `function` keyword; I would reserve using `=>` for unnamed functions or similar in-line definitions.

But, if this is all actually working, I won't push back too hard....